wp-pdf-templates.php

```diff
@@ -207,24 +207,47 @@ function _use_pdf_template() {
 
       if( defined('FETCH_COOKIES_ENABLED') && FETCH_COOKIES_ENABLED ) {
 
-        // we want a html template
-        $header = 'Accept:text/html' . "\n";
-
-        // pass cookies from current request
-        if( isset( $_SERVER['HTTP_COOKIE'] ) ) {
-          $header .= 'Cookie: ' . $_SERVER['HTTP_COOKIE'] . "\n";
-        }
-
-        // create a request context for file_get_contents
-        $context = stream_context_create(array(
-          'http' => array(
-            'method' => 'GET',
-            'header' => $header,
-          )
-        ));
-
-        // load the generated html from the template endpoint
-        $html = file_get_contents($link . (strpos($link, '?') === false ? '?' : '&') . 'pdf-template', false, $context);
+		$url =  $link . (strpos($link, '?') === false ? '?' : '&') . 'pdf-template';
+		$ch = curl_init();
+		
+		curl_setopt($ch, CURLOPT_URL, $url);
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+		
+		// get http headers for cookies
+		curl_setopt($ch, CURLOPT_VERBOSE, 1);
+		curl_setopt($ch, CURLOPT_HEADER, 1);
+		
+		// store the cookies
+		$cookies = array();
+		foreach ($_COOKIE as $key => $value)
+		{
+			if ($key != 'Array')
```

Comment:
Maybe, but I'm still not convinced we should be looping the $_COOKIE array at all. Why does $_SERVER['HTTP_COOKIE'] not work here?